statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -626,6 +831,113 @@ def __init__(self, model):
         # Setup the cache
         self._cache = resettable_cache()
 
+        # Handle covariance matrix calculation
+        if cov_kwds is None:
+                cov_kwds = {}
+        self.get_robustcov_results(cov_type=cov_type, use_self=True,
+                                   **cov_kwds)
+
+    def get_robustcov_results(self, cov_type='opg', **kwargs):
```

Comment:
I had forgotten, privatize this with leading underscore. This does not work for creating a correct results instance if some of the cached attributes are already set.
The `use_self` option only works if none of the cached properties like bse have been set.